src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.Searcher.cs

```diff
@@ -52,34 +52,26 @@ private class Searcher
                 }
             }
 
-            internal void Search()
+            internal async void Search()
             {
-                var navigateToSearch = Logger.LogBlock(FunctionId.NavigateTo_Search, _cancellationToken);
-                var asyncToken = _asyncListener.BeginAsyncOperation(GetType() + ".Search");
-
-                _progress.AddItems(_solution.Projects.Count());
+                try
+                {
+                    using (var navigateToSearch = Logger.LogBlock(FunctionId.NavigateTo_Search, _cancellationToken))
+                    using (var asyncToken = _asyncListener.BeginAsyncOperation(GetType() + ".Search"))
+                    {
+                        _progress.AddItems(_solution.Projects.Count());
 
-                // make sure we run actual search from other thread. and let this thread return to caller as soon as possible.
-                var dummy = Task.Run(() => Search(navigateToSearch, asyncToken), _cancellationToken);
-            }
+                        // Search each project with an independent threadpool task.
+                        var searchTasks = _solution.Projects.Select(
+                            p => Task.Run(() => SearchAsync(p))).ToArray();
```

Comment:
There were tasks count = projects count + 1, but now tasks count = projects count × 2
Why you start new task if you already have async method which returns Task?